File: MATTHEW/leetcode/AIcoding/5.Qwen-3/python3_solutions/2401-longest-nice-subarray.py
from typing import List

class Solution:
    def longestNiceSubarray(self, nums: List[int]) -> int:
        max_len = 0
        bit_mask = 0
        left = 0
        for right in range(len(nums)):
            while (bit_mask & nums[right]) != 0:
                bit_mask ^= nums[left]
                left += 1
            bit_mask |= nums[right]
            max_len = max(max_len, right - left + 1)
        return max_len

File: MATTHEW/leetcode/AIcoding/5.Qwen-3/go_solutions/2401-longest-nice-subarray.go
package main

func longestNiceSubarray(nums []int) int {
    maxLen := 0
    bitMask := 0
    left := 0

    for right, num := range nums {
        for bitMask&num != 0 {
            bitMask ^= nums[left]
            left++
        }
        bitMask |= num
        if right-left+1 > maxLen {
            maxLen = right - left + 1
        }
    }

    return maxLen
}